src/Plugin/GraphQL/DataProducer/Routing/RouteLoad.php

```diff
@@ -95,15 +95,15 @@ public function __construct(
    * @return \Drupal\Core\Url|null
    */
   public function resolve($path, RefinableCacheableDependencyInterface $metadata) {
-    if ($this->redirectRepository) {
+    if ($this->redirectRepository && $redirect = $this->redirectRepository->findMatchingRedirect($path, [])) {
       /** @var \Drupal\redirect\Entity\Redirect|null $redirect */
-      $redirect = $this->redirectRepository->findMatchingRedirect($path, []);
-      if ($redirect) {
-        return $redirect->getRedirectUrl();
-      }
+      $url = $redirect->getRedirectUrl();
```

Comment:
This breaks the type annotation on line 99. The `@var` annotation is not needed for `getRedirectUrl`.

Is it possible to take the `$redirect = ` out of the `if` statement for legibility?

I've opened https://www.drupal.org/project/redirect/issues/3309603 to handle the return type annotation that's incorrect for `findMatchingRedirect`.

Reply:
I can take out `$redirect =` but that gets otherwise ugly. Would you prefer something like this?

 
```
    if ($this->redirectRepository) {
      $redirect = $this->redirectRepository->findMatchingRedirect($path, []);
      if ($redirect){
        $url = $redirect->getRedirectUrl();
      }
    }

    if(empty($url)) {
      $url = $this->pathValidator->getUrlIfValidWithoutAccessCheck($path);
    }

    if ($url && $url->isRouted() && $url->access()) {
      return $url;
    }
```

Reply:
Maybe something along the lines of this is better
```
    if ($this->redirectRepository) {
      $redirect = $this->redirectRepository->findMatchingRedirect($path, []);
    }
    if (!empty($redirect)){
      $url = $redirect->getRedirectUrl();
    }
    else {
      $url = $this->pathValidator->getUrlIfValidWithoutAccessCheck($path);
    }
```

Reply:
Done